lib/elixir/src/elixir_erl_pass.erl

```diff
@@ -631,6 +636,30 @@ rewrite_strategy(erlang, Right, Args) ->
     true -> guard_op;
     false -> none
   end;
+rewrite_strategy(Left, shift, [Struct, Opts | RestArgs]) when
+  Left == 'Elixir.Date';
+  Left == 'Elixir.DateTime';
+  Left == 'Elixir.NaiveDateTime';
+  Left == 'Elixir.Time'
+->
+  case basic_type_arg(Opts) of
+    true ->
+      {Module, Fun} =
+        case (Left == 'Elixir.Date' orelse Left == 'Elixir.Time') of
+          % Time and Date have more restrictive validations than Elixir.Duration
+          true -> {Left, 'new_duration!'};
+          false -> {'Elixir.Duration', 'new!'}
```

Comment:
One option is to define a `__duration__!` on all of them. Your call. :)

Reply:
I hesitated doing so but it feels cleaner indeed. Will do it!

Reply:
Done in [40e8d10](https://github.com/elixir-lang/elixir/pull/13489/commits/40e8d106594bc5bd75dd56616a9fa322aae4b4e6)